### ISSUE TITLE
Support different Storage system for outbox and inbox

### ISSUE THREAD

**Issue by apatozi** [dotnetcore/CAP#1462]
This is a feature request to support different Storage systems for outbox and for inbox. 

There is a scenario for MS where a BFF is used in combination with Websockets. All the instances of the BFF should receive some kind of event in order to push a WS message from the instances that match a WS connection with the criteria. Now all these messages are ephemeral and if someone gets lost, not a big deal. On the other hand every instance should process the same message so every instance should be in its own unique group. If these instances share the same DB then only one will process a message. Let's say with some challenge we overcome the DB table problem and each instance uses its own "Received" and "Published" table (currently only MySql allows table prefix anyway). Now why should we waste resources and latency bothering to use a DB in the BFF instances ? So there is a wise thing to only use in-memory-storage. The problem rises when we want to use Messages of type "Command" that the BFF should send to other services. These need persistent storage to ensure resiliency because the client might send a request for a transaction and the BFF sends this transaction to the other Service and never knows back if the transaction is queued or not because it crashes the instant after. With only the outbox pattern to store to DB and the inbox pattern to use the memory we get the best of both worlds. _I understand that still remains the problem if the above situation happens between the BFF and the WS client but this way we cut down the number of possible failure points_ 

Let me know if this is something that complies with the philosophy of the library and I will make a PR with the changes.

**Comment by yang-xiaodong** [comment]
There's something about your use case that I don't quite understand.

> All the instances of the BFF should receive some kind of event in order to push a WS message from the instances that match a WS connection with the criteria

We're not support websocket messages. What does it mean here?

> On the other hand every instance should process the same message so every instance should be in its own unique group.

Does the `group` here refer to the consumer group in CAP?

> Each instance uses its own "Received" and "Published" table (currently only MySql allows table prefix anyway)

All database are support prefix, SqlServer and Postgres use `Schema` .

---  

Another benefit of the inbox is that you can easily check messages or retrying.

**Comment by apatozi** [comment]
The BFF with Websockets are outside of CAP, those are microservices that have a local state (WS connections) and for this reason all of theirs instances need to subscribe to some topic and all receive the message (they are not competing consumers).  To achieve this CAP has the consumer group property that facilitates this behavior.  In my specific case I don't want to have the inbox for these instances above because it affects performance, I need to do a clean up of DB tables and schemas etc. I can use In-memory storage for this one but that way I loose also the outbox pattern. The outbox doesn't present the problems listed above, it can be shared across instances using the same table/schema. 

![Event](https://github.com/dotnetcore/CAP/assets/17928878/7c87b668-b592-4dcd-be5c-cd1a53334716)

To summarize: Messages from the BFF are important in some cases and must not be lost. Messages to the BFF are not important and no problem if they get lost.

**Comment by yang-xiaodong** [comment]
This is a good use case, but we won't consider it in the short term, as we haven't received such feedback in the past few years. Additionally, it would require significant changes to our current design and code structure.
